extensions/blocks/slideshow/slideshow.php

```diff
@@ -24,5 +24,88 @@
  */
 function jetpack_slideshow_block_load_assets( $attr, $content ) {
 	Jetpack_Gutenberg::load_assets_as_required( 'slideshow' );
+	if ( class_exists( 'Jetpack_AMP_Support' ) && Jetpack_AMP_Support::is_amp_request() ) {
+		return jetpack_slideshow_block_render_amp( $attr );
+	}
 	return $content;
 }
+
+/**
+ * Render slideshow block for AMP
+ *
+ * @param array $attr Array containing the slideshow block attributes.
+ *
+ * @return string
+ */
+function jetpack_slideshow_block_render_amp( $attr ) {
+	global $wp_block_jetpack_slideshow_id;
+	$wp_block_jetpack_slideshow_id = ( $wp_block_jetpack_slideshow_id || 0 ) + 1;
+	$amp_carousel_id               = sprintf(
+		'wp-block-jetpack-slideshow__%s',
+		intval( $wp_block_jetpack_slideshow_id )
+	);
+
+	$ids        = empty( $attr['ids'] ) ? array() : $attr['ids'];
+	$autoplay   = empty( $attr['autoplay'] ) ? false : $attr['autoplay'];
+	$autoplay   = false;
+	$delay      = empty( $attr['delay'] ) ? 3 : intval( $attr['delay'] );
+	$align      = isset( $attr['align'] ) ? $attr['align'] : 'center';
+	$classes    = array(
+		'wp-block-jetpack-slideshow',
+		'wp-amp-block',
+		'align' . $align,
+	);
+	$slides     = array_map(
+		function( $id ) {
+			$caption    = wp_get_attachment_caption( $id );
+			$src        = wp_get_attachment_image_src( $id, 'full' );
+			$figcaption = $caption ? sprintf(
+				'<figcaption class="wp-block-jetpack-slideshow_caption gallery-caption">%s</figcaption>',
+				wp_kses_post( $caption )
+			) : '';
+			$amp_img    = sprintf(
+				'<amp-img src="%s" width="%s" height="%s" alt="%s" class="wp-block-jetpack-slideshow_image" />',
+				esc_url( $src[0] ),
+				esc_attr( $src[1] ),
+				esc_attr( $src[2] ),
+				esc_attr( $caption )
+			);
+			return sprintf(
+				'<div class="wp-block-jetpack-slideshow_slide"><figure>%s%s</figure></div>',
+				$amp_img,
+				$figcaption
+			);
+		},
+		$ids
+	);
+	$bullets    = array_map(
+		function( $index ) use ( $amp_carousel_id ) {
+			return sprintf(
+				'<button class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide %s" on="tap:%s.goToSlide(index=%s)"></button>',
+				( $index + 1 ),
+				esc_attr( $amp_carousel_id ),
+				$index
+			);
+		},
+		array_keys( $ids )
+	);
+	$pagination = sprintf(
+		'<div class="wp-block-jetpack-slideshow_pagination swiper-pagination swiper-pagination-bullets amp-pagination">%s</div>',
+		implode( '', $bullets )
+	);
+	$carousel   = sprintf(
+		'<amp-carousel height="300" width="400" layout="responsive" type="slides" data-next-button-aria-label="%s" data-prev-button-aria-label="%s" controls loop %s id="%s">%s</amp-carousel>',
```

Comment:
Is this `height`/`width` always right?  Would it not be variable based on the images that will be displayed?

Reply:
This is a tricky one, as noted: `Dynamic height sizing will not occur in the AMP version.`. In the non-AMP version, the resizing logic is fairly complex, taking  the available width into account as well as the aspect ratio of the first image: https://github.com/Automattic/jetpack/blob/master/extensions/blocks/slideshow/swiper-callbacks.js#L34-L56

It would be tricky to pull the same off in PHP although I could possibly approximate based on the aspect ratio of the first image. Do AMP's layout settings offer any magic I should try?

Reply:
Actually, dynamically changing the height should be possible via [`amp-bind`](https://amp.dev/documentation/components/amp-bind). You can define a `[height]` attribute on the `amp-carousel` and when changing the slide, update some `amp-state` for the carousel to then populate in  the `height` attribute. This would dynamically change the height based on the currently-selected image.

Reply:
In the non-AMP version the height doesn't change for each slide. It's set once based on the aspect ratio of the first slide, then changes if the width of the block changes. I'm not sure we can replicate this logic without knowing the width of the block, which would only be available client-side.

Reply:
Couldn't the aspect ratio be obtained just by grabbing the first image and then using that image's `width` and `height` as the dimensions for the `amp-carousel`? Note that since you are using the `responsive` layout, it will automatically stretch to take up the full width regardless of the actual dimensions. In this way, the width/height are just used as an aspect ratio. In other words `height="300" width="400"` is identical to `height="3" width="4"`: both result in an aspect ratio of 0.75.

Reply:
This should do it: https://github.com/Automattic/jetpack/pull/13009/commits/3cfa9017b5e166a7c2db9a94426079e76f291fb6. Overall aspect ratio is set based on the first slide, and all others are contained in the available space.